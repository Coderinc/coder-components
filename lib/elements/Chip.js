import React from 'react';
import styled from 'styled-components';

const phases = {
  Develop: {
    "background": "pink",
    "color": "white"
  },
  Measure: {
    "background": "blue",
    "color": "darkGray"
  },
  "Design": {
    "background": "yellow",
    "color": "darkGray"
  },
  "Summarize": {
    "background": "orange",
    "color": "darkGray"
  },
  "Research": {
    "background": "darkBlue",
    "color": "white"
  },
  "Execute": {
    "background": "green",
    "color": "darkGray"
  },
  "Model": {
    "background": "red",
    "color": "white"
  },
  "Test": {
    "background": "purple",
    "color": "white"
  },
  "Maintain": {
    "background": "teal",
    "color": "darkGray"
  }
};


const ChipWrapper = styled.div`
  padding: 0.25rem 0.5rem;
  margin: 0.25rem;
  border-radius: 0.125rem;

  font-size: 0.625rem;

  background-color: ${props => (phases[props.label] ?
    props.theme[phases[props.label].background] : props.theme.primary)};
  color: ${props => 
    (phases[props.label] ?
      props.theme[phases[props.label].color] : props.theme.primaryFont)};
`;

export const Chip = props => (
  <ChipWrapper {...props}>{props.label}</ChipWrapper>
);

export default Chip;
