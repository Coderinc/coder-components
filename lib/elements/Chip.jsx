import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const phases = {
  Develop: {
    background: 'pink',
    color: 'white',
  },
  Measure: {
    background: 'blue',
    color: 'darkGray',
  },
  Design: {
    background: 'yellow',
    color: 'darkGray',
  },
  Summarize: {
    background: 'orange',
    color: 'darkGray',
  },
  Research: {
    background: 'darkBlue',
    color: 'white',
  },
  Execute: {
    background: 'green',
    color: 'darkGray',
  },
  Model: {
    background: 'red',
    color: 'white',
  },
  Test: {
    background: 'purple',
    color: 'white',
  },
  Maintain: {
    background: 'teal',
    color: 'darkGray',
  },
};

const ChipWrapper = styled.div`
  padding: 0.25rem 0.5rem;
  margin: 0.25rem;
  border-radius: 0.125rem;

  font-size: 0.625rem;

  background-color: ${({ label, theme }) => (phases[label] ? theme[phases[label].background] : theme.primary)};
  color: ${({ label, theme }) => (phases[label] ? theme[phases[label].color] : theme.primaryFont)};
`;

const Chip = ({ label, ...props }) => (
  <ChipWrapper {...props} label={label}>
    {label}
  </ChipWrapper>
);

Chip.propTypes = {
  label: PropTypes.string,
};

Chip.defaultProps = {
  label: 'Phase',
};

export default Chip;
