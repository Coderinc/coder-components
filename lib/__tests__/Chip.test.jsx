import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Chip from '../elements/Chip';

describe('Chip', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Chip label="Develop" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('chooses correct color', () => {
    const component = renderer.create(<Chip theme={{ primary: '#FF0000' }} label="test" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
