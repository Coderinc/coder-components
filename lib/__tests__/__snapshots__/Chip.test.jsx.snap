// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip chooses correct color 1`] = `
.c0 {
  padding: 0.25rem 0.5rem;
  margin: 0.25rem;
  border-radius: 0.125rem;
  font-size: 0.625rem;
  background-color: #FF0000;
}

<div
  className="c0"
  label="test"
>
  test
</div>
`;

exports[`Chip renders correctly 1`] = `
.c0 {
  padding: 0.25rem 0.5rem;
  margin: 0.25rem;
  border-radius: 0.125rem;
  font-size: 0.625rem;
}

<div
  className="c0"
  label="Develop"
>
  Develop
</div>
`;
